### PR TITLE
ISPN-7955 Make the Hot Rod client lazily resolve server addresses

### PR DESCRIPTION
**Pull request by tristantarrant**
https://issues.jboss.org/browse/ISPN-7955

### CODE REVIEW

**Comment by tristantarrant** [comment]
run performance tests please
cs-rest

**Comment by tristantarrant** [comment]
run performance tests please
cs-hotrod

**Comment by tristantarrant** [comment]
The change has no impact on performance

**Comment by tristantarrant** [comment]
The user on the forum has reported success with this code

File: client/hotrod-client/src/test/java/org/infinispan/client/hotrod/ConsistentHashV2Test.java [hunk 1]
```diff
@@ -2,6 +2,7 @@
 
 import static org.testng.Assert.assertEquals;
 
+import java.net.InetAddress;
```
Comment (wburns) [inline client/hotrod-client/src/test/java/org/infinispan/client/hotrod/ConsistentHashV2Test.java]: Unused.

**Comment by wburns** [comment]
It seems fine to me but there appear to be a bunch of possibly related CI failures.

http://ci.infinispan.org/job/Infinispan/job/PR-5223/5/

**Comment by wburns** [comment]
@tristantarrant it seems I cannot integrate this. As far as I know there were failures in a previous build on CI. Did you mean to push some changes?  I thought you mentioned you had this fixed.

**Comment by tristantarrant** [comment]
There was one last failure I could not fix, so I'll get it done for final

**Comment by wburns** [comment]
@tristantarrant Thanks, moved it over.

**Comment by tristantarrant** [comment]
Fixed the failing tests.

**Comment by wburns** [comment]
Integrated into master, thanks @tristantarrant !

**Pull request merged**